src/include/server/mir/shell/decoration.h

```diff
@@ -0,0 +1,7 @@
+#include "mir/geometry/forward.h"
+#include "mir_toolkit/common.h"
+
+namespace mir::shell::decoration
+{
+    auto compute_size_with_decorations(geometry::Size content_size, MirWindowType type, MirWindowState state) -> geometry::Size;
+}
```

Comment:
Might not be the best way to expose `compute_size_with_decorations` to xwayland code, suggestions welcome

Reply:
This will cause issues when we come to expose a configuration point for decorations. But good for now.

Reply:
What's another better approach I could try?

Reply:
This is a good approach (for now). Don't make things complicated to solve problems we haven't encountered yet.